imagebuildah/stage_executor.go

```diff
@@ -590,10 +615,60 @@ func (s *StageExecutor) runStageMountPoints(mountList []string) (map[string]inte
 	return stageMountPoints, nil
 }
 
+func (s *StageExecutor) CreateNeededHeredocMountsForRun(files []imagebuilder.File) ([]Mount, error) {
+	mountResult := []Mount{}
+	for _, file := range files {
+		fileName := file.Name
+		if file.Name == "" || file.Name == "EOF" {
+			fileName = fmt.Sprintf("%s-%d", "buildahheredoc", rand.Intn(10000))
+		}
+		f, err := ioutil.TempFile("", fileName)
+		if err != nil {
+			return nil, err
+		}
+		if _, err := f.WriteString(file.Data); err != nil {
+			return nil, err
+		}
+		err = os.Chmod(f.Name(), 0711)
+		if err != nil {
+			return nil, err
+		}
+		// dest path is same as buildkit for compat
+		dest := filepath.Join("/dev/pipes/", fileName)
+		mount := Mount{Destination: dest, Type: "rbind", Source: f.Name(), Options: []string{"z"}}
+		mountResult = append(mountResult, mount)
+		f.Close()
+	}
+	return mountResult, nil
+}
+
 // Run executes a RUN instruction using the stage's current working container
 // as a root directory.
 func (s *StageExecutor) Run(run imagebuilder.Run, config docker.Config) error {
 	logrus.Debugf("RUN %#v, %#v", run, config)
+	args := run.Args
+	if len(run.Files) > 0 {
+		if heredoc := buildkitparser.MustParseHeredoc(args[0]); heredoc != nil {
```

Comment:
We are only using this here to detect if more than one heredoc is present or not, if there is any error in heredoc syntax imagebuilder fails much early with needed error so handling error here does will not have any effect since imagebuilder will fail early. I can add a test for bad case to verify this in CI 
```
Error: parsing main Dockerfile: /work/Dockerfile: EOF: unterminated heredoc
```